### ISSUE TITLE
SCS Infiltration Calculator

### ISSUE THREAD

**Issue by FLO-2DKaren** [FLO-2DSoftware/qgis-flo-2d-plugin#768]

**Comment by ldebek** [comment]
@FLO-2DKaren Changes are ready for testing on branch: https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/tree/oct2021_sprint

**Comment by FLO-2DKaren** [comment]
@ldebek I'll test it soon but I have a potential issue.  I just found the Curve Number Generator plugin and it suits our plugin very nicely.  I may want to add some programming related to that so we can put it on another job or something.  I'll get this tested and then meet with you.

**Comment by FLO-2DJJ** [comment]
@FLO-2DKaren can this issue be closed?

**Comment by FLO-2DKaren** [comment]
No I want to do something a little different.  I'll get to this soon.

**Comment by FLO-2DKaren** [comment]
@FLO-2DJJ 
I still think there is a need for calculating scs infiltration from a raster.  This is because the scs calculator kind of sucks.  I don't know why it can't deal with the slightest conflict.  The scs from raster code is here: https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/tree/oct2021_sprint

However, I don't like that code because it eliminates one of the calculators.  I think we should offer both calculators.  The Curve Number Generator works really well and therefore the normal SCS calculator works really well.  We still need the raster calculator too.  Let's discuss on Friday.

**Comment by FLO-2DKaren** [comment]
Add a raster SCS calculator to this window.
![image](https://user-images.githubusercontent.com/20424460/212379413-9db437d9-0009-47c3-8534-1918fe651be0.png)

You can get the code from the sample grid element elevation and modify it for SCS.
Final note on this is that CN needs to be an Integer.
![image](https://user-images.githubusercontent.com/20424460/212379485-9606ff3a-30e9-4011-9d74-475c9a5c92fa.png)

**Comment by rpachaly** [comment]
I've finished the new SCS Curve Number raster calculator. Here is an image of the final SCS Calculator:

![Image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/39889306/4d37076b-3335-4d72-b13e-0190093b5b12)

My testing procedure was the following:

- Created two raster containing the CN values, one with missing data and other complete
- Ran multiple combinations of resampling algorithm, fill nodata checked or unchecked, and many other combinations. No error was found

The final CN value added to the "Cells SCS" is an integer. 

I also added the documentation for the SCS raster calculator and adjusted the other two methods for matching the new SCS Calculator UI. 

Obs: I couldn't open the links above, so I coded everything based on the Roughness Sampling from raster.

I'm pushing the branch right now.
